pycqed/measurement/multi_qubit_module.py

```diff
@@ -304,22 +301,32 @@ def get_multi_qubit_prep_params(prep_params_list):
     return prep_params
 
 
-def get_meas_obj_value_names_map(qubits, det_type):
-    if det_type == 'int_avg_classif_det':
+def get_meas_obj_value_names_map(mobjs, multi_uhf_det_func):
+    # we cannot just use the value_names from the qubit detector functions
+    # because the UHF_multi_detector function adds suffixes
+
+    if multi_uhf_det_func.detectors[0].name == 'raw_UHFQC_classifier_det':
         meas_obj_value_names_map = {
-            qb.name: [vn + ' ' + qb.instr_uhf() for
-                      vn in qb.int_avg_classif_det.value_names]
-            for qb in qubits}
-    elif det_type == 'inp_avg_det':
+            qb.name: hlp_mod.get_sublst_with_all_strings_of_list(
+                multi_uhf_det_func.value_names,
+                qb.int_avg_classif_det.value_names)
+            for qb in mobjs}
+    elif multi_uhf_det_func.detectors[0].name == 'UHFQC_input_average_detector':
         meas_obj_value_names_map = {
-            qb.name: [vn + ' ' + qb.instr_uhf() for
-                      vn in qb.inp_avg_det.value_names]
-            for qb in qubits}
+            qb.name: hlp_mod.get_sublst_with_all_strings_of_list(
+                multi_uhf_det_func.value_names, qb.inp_avg_det.value_names)
+            for qb in mobjs}
     else:
         meas_obj_value_names_map = {
-            qb.name: [vn + ' ' + qb.instr_uhf() for
-                      vn in qb.int_avg_det.value_names]
-            for qb in qubits}
+            qb.name: hlp_mod.get_sublst_with_all_strings_of_list(
+                multi_uhf_det_func.value_names, qb.int_avg_det.value_names)
+            for qb in mobjs}
+
+    meas_obj_value_names_map.update({
+        name: [name] for name in
+        [vn for vn in multi_uhf_det_func.value_names if vn not in
+         hlp_mod.flatten_list(list(meas_obj_value_names_map.values()))]})
+
```

Comment:
I can't judge if this will break any existing analysis.
What have you tested this on?

Reply:
@antsr  "meas_obj_value_names_map" and hence this method is only used for analysis_v3. I wrote this helper function for better integration of measurements into analysis_v3. This helper function is currently only used in measure_two_qubit_randomized_bechmarking.

Reply:
Also I was tested on BF1 together with @MichKe95 .